### ISSUE TITLE
Change the response result when error occurs, like redirecting to another method

### ISSUE THREAD

**Issue by zealseeker** [strongloop/loopback#2706]
The requirement is similar to #634. But here is my condition:
When I POST a new user in user model, it will show an error if the username exists.
I want to change the password of the existed instance into the new password I posted.

1) I can do this in beforeRemote('create') and the password will be changed successfully. However, the response will show the error of "existed username" because I didn't create a new user but only changed the old one. According to the mechanism shown in #634, it seems that I cannot change the response.
Is it possible to change it ? I want to make it successful and return the "new user", instead of showing the error.

2) I can do this in afterRemoteError also, but the problem still exists. Whatever I do in this remote hook, I cannot change the truth that it is an error.

In conclusion, I want to make the method 'create' become 'update' in a specific condition (the username exists) and return successful as I have created it.

It is available to modify the client side. But it will lead to compatibility problem. So it is the least way to change the client codes.

**Comment by zealseeker** [comment]
Finally I solved my problem with the following code:
```js
User.beforeRemote('create', function (ctx, unused, next) {
    User.findOne({where:{username:username}},function(err,user){
          if(err || !user){
            next();
          } else {
            if (typeof(user.IsNoPassword)!=='undefined' && user.IsNoPassword){
              user.password = ctx.req.body.password;
              user.IsNoPassword = false;
              user.save(function(err,instance){
                ctx.res.send(instance);
              })
            } else {
              next();
            }
          }
    });
});
```
I use `IsNoPassword` to label whehter a user can change his password by the remote method of create.

`res.send` is the key to implement my idea.  But it does not satisfy the feature requirement I posted.
Hope that we can redirect the remote method, for example, from create to update or other custom remote method. So that we need not to pay attention to the output format becuase it is just handled by another remote method workflow.

**Comment by jannyHou** [comment]
@zealseeker 
- From my understanding, https://github.com/strongloop/loopback/issues/634 is for the situation that you want to change response result for **ALL** models, but in your case you only want to change `User` model.

- I am a little confused with your requirement, to be honest I don't think it's a good idea to change password in `beforeRemoting` hook, please read the suggestion from our [doc](https://docs.strongloop.com/display/public/LB/Remote+hooks):
> Use beforeRemote hooks to validate and sanitize inputs to a remote method. Because a beforeRemote hook runs before the remote method is executed, it can access the inputs to the remote method, but not the result.
Use afterRemote hooks to modify, log, or otherwise use the results of a remote method before sending it to a remote client. Because an afterRemote hook runs after the remote method is executed, it can access the result of the remote method, but cannot modify the input arguments.

- So could you clarify your use case in terms of what's your requirements?
- Also, please notice that we have operation hooks(see [doc](https://docs.strongloop.com/display/public/LB/Operation+hooks) for it), in which case you don't have to redirect the remote method, maybe a `beforesave` hook is what you need.

**Comment by zealseeker** [comment]
@jannyHou Thanks for the suggestion. In my case, it is better to change password in 'AfterRemoteError'. But my requirement is to "Automatically change remote method"

For example, I have two remote methods, Remote_One (create a user) and Remote_Two (change the user's password).   `User.property : [username, password, IsNoPassword]`  When a user's `IsNoPassword==true`, which means he has registed without setting password, we allow the user regist again.
In Loopback, if the user regist again, he will get the result of "existed username". So I want to add a beforeRemote hook: when we find that the user's `IsNoPassword==true`, we change the RemoteMethod from RemoteOne to RemoteTwo.

With beforeRemote hook, I can execute Remote_two (setting the password), but it won't return the result of Remote_two (the instance after update) . So what I want is redirect the remote method, where the workflow of Remote_One will break and finally return the result of remote_two.

Besides, is there any function that can pretend remote method, which will call another remote hook?
